### PR TITLE
refactor: Converted file scan to RX operation

### LINKED ISSUE #661: Refactor: Convert File scan thread into RX operations
#### Feature Description

Right now we start separate threads in the ```scan``` method of the ```FileSearch``` class in order to do custom file search and media file search [here](https://github.com/kiwix/kiwix-android/blob/108af1096b9316f9dd00ecee912aff4c5cfc4942/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileSearch.java#L62-L73).
These should be converted to RX Observable.

Parent issue: #176

### PR DESCRIPTION
**Pull request by harshika-arya**
Fixes #661 

Changes: Converted file scan operation to use Rx completable. Hence removing the need of the `checkCompleted` method.

Reasons for the change: RxJava Completables are completely managed by the library. As a result, we don't need to check for completion after one thread finishes(which in turn meant that we had to synchronize the access). The library will itself wait for both the threads to finish and then run whatever is mentioned in the `doOnCompleted` method. Moreover, the code becomes concise and readable.

Screenshots/GIF for the change: N/A

@abdulwd @siddharth2010 @mhutti1 @soloturn Please review.

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileSearch.java [hunk 1]
```diff
@@ -92,27 +96,27 @@ public FileSearch(Context ctx, ResultListener listener) {
     return book;
   }
 
+  /**
+   * Uses RxJava Completable to scan the file system and media store concurrently.
+   * Synchronizes the check and finally calls onScanCompleted after completion.
+   * @param defaultPath
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileSearch.java]: You should either explain this parameter or remove this line.

Comment (harshika-arya) [reply app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileSearch.java]: Ok
Any clue on the build failure? I can't see any stacktrace on Travis CI.

Comment (abdulwd) [reply app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileSearch.java]: We run some Espresso tests on Tesdroid. On some device(s) one or more tests have failed. You should always run Espresso test on your local devices before making a PR.

**Comment by soloturn** [comment]
thank you so much @harshika-arya, i like the change! git version control gives you the possibility to:
* squash commits
* ammend to last commit
* edit commit message
* rebase master and force push
it makes it a lot easier to review if the PR is less commits, and the commit message includes why you changed something. so - if you could squash and rebase yours would be much appreciated.

**Comment by harshika-arya** [comment]
@soloturn Thanks, I have rebased and added some info both in the commit message as well as in the PR.

**Comment by harshika-arya** [comment]
@abdulwd @soloturn This is ready for review.

**Comment by kelson42** [comment]
@harshika-arya @abdulwd @macgills What should we do with PR?

**Comment by macgills** [comment]
@harshika-arya thanks so much for your contribution, unfortunately I had much a similar idea in #1238 so this has been done.
We hope to see more contributions from you in the future

**Pull request closed without merging**